File: python/docs_code_snippets/functions-create.py
import time
from opperai import Opper

opper = Opper()

fn_name = f"docs-snippet-{int(time.time())}"

# --- docs ---
# Functions are auto-created on first call
result = opper.call(
    fn_name,
    input={"text": "Hello world"},
    input_schema={"type": "object", "properties": {"text": {"type": "string"}}},
    output_schema={"type": "object", "properties": {"summary": {"type": "string"}}},
)
print(result.data)
# --- /docs ---

# Cleanup
opper.functions.delete(fn_name)

File: typescript/docs_code_snippets/functions-create.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

// Functions are auto-created on first call
const result = await opper.call("my-new-function", {
  input: { text: "Hello world" },
  input_schema: z.object({ text: z.string() }),
  output_schema: z.object({ summary: z.string() }),
});
console.log(result.data);
